test/unit/uk/gov/hmrc/tariffclassificationfrontend/controllers/SampleControllerSpec.scala

```diff
@@ -0,0 +1,245 @@
+import org.scalatest.mockito.MockitoSugar
+import org.scalatest.{BeforeAndAfterEach, Matchers}
+import play.api.http.Status
+import play.api.i18n.{DefaultLangs, DefaultMessagesApi}
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import play.api.{Configuration, Environment}
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
+import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+import uk.gov.hmrc.tariffclassificationfrontend.models.Permission.Permission
+import uk.gov.hmrc.tariffclassificationfrontend.models.SampleStatus.SampleStatus
+import uk.gov.hmrc.tariffclassificationfrontend.models._
+import uk.gov.hmrc.tariffclassificationfrontend.service.CasesService
+import uk.gov.tariffclassificationfrontend.utils.Cases._
+
+import scala.concurrent.Future
+
+class SampleControllerSpec extends UnitSpec with Matchers
+  with WithFakeApplication with MockitoSugar with ControllerCommons with BeforeAndAfterEach {
+
+  private val fakeRequest = FakeRequest()
+  private val env = Environment.simple()
+  private val configuration = Configuration.load(env)
+  private val messageApi = new DefaultMessagesApi(env, configuration, new DefaultLangs(configuration))
+  private val appConfig = new AppConfig(configuration, env)
+  private val casesService = mock[CasesService]
+  private val operator = mock[Operator]
+
+  private def controller(requestCase: Case) = new SampleController(
+    new SuccessfulRequestActions(operator, c = requestCase), casesService, messageApi, appConfig
+  )
+
+  private def controller(requestCase: Case, permission: Set[Permission]) = new SampleController(
+    new RequestActionsWithPermissions(permission, c = requestCase), casesService, messageApi, appConfig
+  )
+
+  private implicit val hc: HeaderCarrier = HeaderCarrier()
+
+  override def afterEach(): Unit = {
+    super.afterEach()
+    Mockito.reset(casesService)
+  }
+
+  "Sample controller" should {
+
+    "return 200 OK and HTML content type - For CANCELLED Case" in {
+      val c = aCase(withStatus(CaseStatus.CANCELLED), withDecision())
+
+      val result = await(controller(c).sampleDetails("reference")(fakeRequest))
+
+      status(result) shouldBe Status.OK
+      contentType(result) shouldBe Some("text/html")
+      charset(result) shouldBe Some("utf-8")
+      contentAsString(result) should include("sample-status-heading")
+    }
+
+    "return 200 OK and HTML content type - For COMPLETED Case" in {
+      val c = aCase(withStatus(CaseStatus.COMPLETED), withDecision())
+
+      val result = await(controller(c).sampleDetails("reference")(fakeRequest))
+
+      status(result) shouldBe Status.OK
+      contentType(result) shouldBe Some("text/html")
+      charset(result) shouldBe Some("utf-8")
+      contentAsString(result) should include("sample-status-heading")
+    }
+
+    "return 200 OK and HTML content type - For OPEN Case" in {
+      val c = aCase(withStatus(CaseStatus.OPEN), withDecision())
```

Comment:
Is there any significance to the case status in these first 3 tests?  Tests are all asserting the same thing

Reply:
Actually no, since we allow it in any status, I should update this

Reply:
Have updated this one and removed irrelevant repeats